src/Internal/ProcessConstraints.purs

```diff
@@ -574,19 +578,20 @@ checkRefNative
   :: forall (a :: Type)
    . InputWithScriptRef
   -> ConstraintsM a (Either MkUnbalancedTxError Boolean)
-checkRefNative scriptRef = pure $ note (WrongRefScriptHash Nothing) $ isNative
-  (unwrap (unwrap uout).output).scriptRef
+checkRefNative scriptRef =
+  case scriptRef of
+    RefInput ref -> isNative ref
+    SpendInput ref -> isNative ref
+    RefInputUnchecked _ -> pure $ Right false
```

Comment:
This is not going to work. It could as well be native.

Reply:
Made it so that when `RefInputUnchecked` is given, it will just bypass the native script check altogether